crates/astria-composer/src/composer.rs

```diff
@@ -90,3 +165,45 @@ fn report_exit(task_name: &str, outcome: Result<eyre::Result<()>, JoinError>) {
         }
     }
 }
+
+#[async_trait::async_trait]
+impl ComposerService for ExecutorHandle {
+    async fn submit_sequence_actions(
+        &self,
+        request: Request<SubmitSequenceActionsRequest>,
+    ) -> Result<Response<()>, tonic::Status> {
+        let submit_sequence_actions_request = request.into_inner();
+        if submit_sequence_actions_request.sequence_actions.is_empty() {
+            return Err(tonic::Status::invalid_argument(
+                "No sequence actions provided",
+            ));
+        }
+
+        // package the sequence actions into a SequenceAction and send it to the searcher
+        for sequence_action in submit_sequence_actions_request.sequence_actions {
+            let sequence_action = SequenceAction {
+                rollup_id: RollupId::from_unhashed_bytes(sequence_action.rollup_id),
+                data: sequence_action.tx_bytes,
+                fee_asset_id: default_native_asset_id(),
+            };
+
+            match self
+                .send_bundles
+                .send_timeout(sequence_action, Duration::from_millis(500))
+                .await
+            {
+                Ok(()) => {}
+                Err(SendTimeoutError::Timeout(_seq_action)) => {
+                    return Err(tonic::Status::deadline_exceeded(
+                        "timeout while sending txs to searcher",
+                    ));
+                }
+                Err(SendTimeoutError::Closed(_seq_action)) => {
+                    return Err(tonic::Status::unavailable("searcher is not available"));
+                }
+            }
+        }
+
```

Comment:
question: is the intention here to send all the actions within the rpc request as a bundle to be executed atomically? because right now it's sending individually, which should probably be documented clearer in the grpc proto definition if that is the intended usage. however it seems there should be a way to send a bunch of actions as a bundle?

Reply:
thats a good question. I believe this can be taken as a future enhancement for now. will discuss with Itamar.